contracts/near/eth-client/src/lib.rs

```diff
@@ -348,12 +383,192 @@ impl EthClient {
         }
     }
 
-    /// Verify PoW of the header.
     fn verify_header(
         &self,
         header: &BlockHeader,
         prev: &BlockHeader,
         dag_nodes: &[DoubleNodeWithMerkleProof],
+    ) -> bool {
+        // if the validate_header is false, header are not validated. 
+        if self.validate_header == false {
```

Comment:
So if you look at the original code, if `validate_header` is set to false, there is still some verification done on the header fields. So maybe we can split header verification into three parts:
- Basic verification which is done regardless of `validate_header` and `validate_header_mode` flags. Basic verification would include all checks that would work on both chains.
 - Then we check whether `validate_header` flag
 - And then finally, we do final check depending on which algorithm to use, ethash or parlia.

Thoughts?